pumpkin-lib/src/engine/constraint_satisfaction_solver.rs

```diff
@@ -151,9 +151,12 @@ pub struct ConstraintSatisfactionSolver {
     /// of the clausal propagator.
     watch_list_propositional: WatchListPropositional,
     /// Used in combination with the propositional watch list
-    /// Indicates the next literal on the propositional trail that need to be inspected to notify
-    /// subscribed propagators.
+    /// Indicates the next literal on the propositional trail that needs to be inspected to notify
+    /// subscribed propagator(s).
     propositional_trail_index: usize,
+    /// Indicates the next entry on the CP trail that needs to be inspected to notify the
+    /// subscribed propagator(s).
+    cp_trail_index: usize,
```

Comment:
The name is a little vague, this is really only used for backtracking events so a more descriptive field name would be more clear.